Framework/Algorithms/src/ConvertUnitsUsingDetectorTable.cpp

```diff
@@ -618,9 +617,10 @@ API::MatrixWorkspace_sptr ConvertUnitsUsingDetectorTable::removeUnphysicalBins(
       auto &X = workspace->x(i);
       auto &Y = workspace->y(i);
       auto &E = workspace->e(i);
-      result->mutableX(i) = HistogramX(X.begin() + first, X.end());
-      result->mutableY(i) = HistogramY(Y.begin() + first, Y.end());
-      result->mutableE(i) = HistogramE(E.begin() + first, E.end());
+
+      result->mutableX(i) = std::move(HistogramX(X.begin() + first, X.end()));
+      result->mutableY(i) = std::move(HistogramY(Y.begin() + first, Y.end()));
+      result->mutableE(i) = std::move(HistogramE(E.begin() + first, E.end()));
```

Comment:
The `std::move` is redundant, you are assigning an rvalue anyway. Also, I think we should really add `FixedLengthVector::assign`.